### PR TITLE
Add a route to catch "404 Not Found" errors

### PR DESCRIPTION
**Pull request by AntoninoM90**

### CODE REVIEW

**Comment by l0gicgate** [comment]
There is no need for this. Slim throws HttpNotFoundException if no route is matched when routing has been performed.

**Comment by eugene-borovov** [comment]
[CORS Pre-Flight route](https://github.com/slimphp/Slim-Skeleton/blob/52d61e0f7d5af6d52570620ed448ef9a23b43b05/app/routes.php#L12) break Slim default behavior. Now you get `405 Method not allowed. Must be one of: OPTIONS` instead of `404 Not found` if no route is matched.

File: app/routes.php [hunk 1]
```diff
@@ -23,4 +24,12 @@
         $group->get('', ListUsersAction::class);
         $group->get('/{id}', ViewUserAction::class);
     });
+
+    /**
+     * Catch-all route to serve a 404 Not Found page if none of the routes match
+     * NOTE: make sure this route is defined last
+     */
+    $app->map(['GET', 'POST', 'PUT', 'DELETE', 'PATCH'], '/{routes:.+}', function (Request $request, Response $response) {
```
Comment (eugene-borovov) [inline app/routes.php:32]: You can use `$app->any('/{routes:.+}')`

**Comment by carlosmintfan** [comment]
@AntoninoM90 I guess @l0gicgate didn't think about that behaviour which @eugene-borovov explained, you could make a new PR. But I'm also not very sure about this catch either because then you'd get 404 instead of 405 for other routes where you really should get 405. So I think I'd make some checking and open a new PR here and for the docs which checks if it return 404 or 405 depending on if the allowed route is OPTIONS or something else.

**Pull request closed without merging**